images/flux/tests/main.tf

```diff
@@ -1,20 +1,58 @@
 terraform {
   required_providers {
-    oci = { source = "chainguard-dev/oci" }
+    oci  = { source = "chainguard-dev/oci" }
+    helm = { source = "hashicorp/helm" }
   }
 }
 
-variable "digest" {
-  description = "The image digest to run tests over."
+variable "digests" {
+  description = "The image digests to run tests over."
+  type = object({
+    cli                         = string
+    helm-controller             = string
+    kustomize-controller        = string
+    notification-controller     = string
+    source-controller           = string
+    image-automation-controller = string
+    image-reflector-controller  = string
+  })
 }
 
-data "oci_exec_test" "help" {
-  digest = var.digest
-  script = "${path.module}/01-help.sh"
+data "oci_string" "ref" {
+  for_each = var.digests
+  input    = each.value
 }
 
-# TODO(imjasonh): Uncomment when we have a KinD cluster.
-# data "oci_exec_test" "flux-export" {
-#   digest = var.digest
-#   script = "${path.module}/02-flux-export.sh"
-# }
+resource "helm_release" "flux" {
+  name             = "flux"
+  namespace        = "flux-system"
+  repository       = "https://fluxcd-community.github.io/helm-charts"
+  chart            = "flux2"
+  create_namespace = true
+
+  values = [
+    <<EOF
+cli:
+  image: ${data.oci_string.ref["cli"].registry_repo}
+  tag: ${data.oci_string.ref["cli"].pseudo_tag}
+helmController:
```

Comment:
Is there a reason we do it this way vs. the other way?  If so, a comment would be helpful to explain why.

generally for structured stuff like this I like to use `jsonencode({...})` since json is valid yaml and it let's us shed the `${}` wrapping, and gets some measure of syntax prevalidation.

Reply:
no good reason, old habbits die hard 😄 

although you got me experimenting with `jsonencode` and I like that better in this case.

I'll land this as is for now, and follow up with a PR shortly after to convert these over alongside the shift to `data.oci_string`

Reply:
nevermind, ran into a merge conflict so changing it here 🎉